mqtt/tls-connect-to-broker.md

```diff
@@ -0,0 +1,128 @@
+
+Alter the <code class="node">MQTT Config</code> node, changing the "Server" name to start with `mqtts://`.
+
+#### Example
+
+**Valid Certificate Creation**
+
+THe easiest approach for this is to use [Let's Encrypt](https://letsencrypt.org/getting-started/). This is beyond the scope of this article but there are plenty of examples and tutorials available on the Internet. For this to work successfully, you also need to be able to use a registered domain name on your internal network because you cannot use Let's Encrypt with IP addresses or non-public domain names.
+
+Alternatively, you can create a self-signed set of certificates. Again, this is beyond the scope of the article. However, you may need to create a trusted root certificate and provide its public cert instead of the one that Debian provides that is listed in the configuration below.
+
+**Mosquitto configuration**
+
+This assumes that you are using Let's Encrypt or other certificates signed by a root CA already trusted by the Debian operating system.
+
+Note the entries in `<...>` which need to be replaced with your own folders and files.
+
+For Linux installations, this goes into a file of any name of the form `*.conf` in the folder `/etc/mosquitto/conf.d/`. So you have to edit it with root privalages (e.g. using `sudo`). On other platforms, please refer to the [Mosquitto configuration documentation](https://mosquitto.org/man/mosquitto-8.html).
+
+{{ page.lcb }}% raw %}
+~~~text
+# Default Listener: 1883
+port 1883
+# Bind the default listener to localhost only if you want to force external connections to be TLS only
+#bind_address localhost
+
+# Secure listener
+listener 8883
+# TLS
+## This is standard and should always be this when using Let's Encrypt
+## If using a self-signed certificate, this needs to be your custom Root CA public certificate
+cafile /etc/ssl/certs/DST_Root_CA_X3.pem
+## These are from your installation of LE
+certfile /<path-to-LE-cert-files>/<fullchain>.cer
+keyfile /<path-to-LE-cert-files>/<private-key-name>.key
+## Forces use of modern version of TLS to avoid security issues
+tls_version tlsv1.2
+
+## Forces ALL CLIENTs using this port to provide a valid certificate - change the node config to allow this from NR
+#require_certificate true
+~~~
+{: .shell}
+{{ page.lcb }}% endraw %}
+
+After making these changes, you have to restart the mosquitto broker. On Linux, you can usr the command:
+
+~~~text
+[~]$ sudo systemctl restart mosquitto
+~~~
+{: .shell}
+
+Other platforms, including Docker-based installations may be different.
+
+**<code class="node">MQTT Config</code> node configuration**
+
+![](/images/mqtt/tls-connect-1.png)
+
+Notes
+
+* You need to use the IP name rather than IP address in the server name if using Let's Encrypt (otherwise the certificate isn't valid).
+* You need to change the server name to a url, prefixed with `mqtts://`. 
+  
+  This disables the port field, I change that first to `8883` to remind me what the correct port will be.
```

Comment:
Another advantage of ticking the "Enable secure connection" checkbox: Because there is no need to add the protocol to the Server field, the port field stays enabled.